src/lib.rs

```diff
@@ -61,6 +61,64 @@ pub use crate::util::{CastFromPrimitive, Pixel};
 
 pub use crate::frame::Frame;
 
+/// Version information
+///
+/// The information is recovered by `cargo` and `git describe` when available.
+pub mod version {
+  /// Major version component
+  ///
+  /// It is increased every time a release presents a incompatible API change.
+  pub fn major() -> usize {
+    env!("CARGO_PKG_VERSION_MAJOR").parse().unwrap()
+  }
+  /// Minor version component
+  ///
+  /// It is increased every time a release presents new functionalities are added
+  /// in a backwards-compatible manner.
+  pub fn minor() -> usize {
+    env!("CARGO_PKG_VERSION_MINOR").parse().unwrap()
+  }
+  /// Patch version component
+  ///
+  /// It is increased every time a release provides only backwards-compatible bugfixes.
+  pub fn patch() -> usize {
+    env!("CARGO_PKG_VERSION_PATCH").parse().unwrap()
+  }
+
+  /// Version information as presented in CARGO_PKG_VERSION
+  ///
+  /// Can be parsed by [semver](https://crates.io/crates/semver).
```

Comment:
Markdown in headers is gross.

Reply:
If I recall correctly it is encouraged to use Markdown in comments because `rustdoc` can parse it. Personally, I agree that it is ugly, but I can see the case for it.

Reply:
Once it is rendered is lovely thought :) (most people use either `cargo doc --open` or docs.rs)